src/visitor.rs

```diff
@@ -584,6 +596,33 @@ impl<'b, 'a: 'b> FmtVisitor<'a> {
 
     // Returns true if we should skip the following item.
     pub fn visit_attrs(&mut self, attrs: &[ast::Attribute], style: ast::AttrStyle) -> bool {
+        for attr in attrs {
```

Comment:
Checking for deprecated `rustfmt_skip` or invalid scoped attributes inside `visit_attrs` means that we cannot emit warnings for attributes on expressions (we use `contains_skip` for expressions).

It would be nice if we can perform this check inside `contains_skip`. I think we can do so by adding `RefCell<FormatReport>` field to `RewriteContext` and pass `&RewriteContext` to `contains_skip`.

Reply:
I was hesitant to do this because there is no guarantee that `contains_skip` will be called only once per attribute and so we potentially end up having to deduplicate errors. Attributes on expressions are also unstable and not widely used, so I don't think it matters too much.